### ISSUE TITLE
Outputs of the last layer and classification

### ISSUE THREAD

**Issue by CatcherInThePy** [dlutkaka/DeepHSV#6]
Hello,

I am trying to get more insight about how the classification part is working and I would be grateful if you could help me about this topic.

When using the `Inception_2logits` model, first we get two embedding values for each input image  in [_loss_inception_2logits](https://github.com/dlutkaka/DeepHSV/blob/master/models.py#L122) function, which actually runs the inputs through your proposed model. Then, in [ _embedding_2logits](https://github.com/dlutkaka/DeepHSV/blob/b39f40c7545129fe5a5b6d19cc298d440d1c0bef/models.py#L54) function we get the concatenated embeddings and give them to a fully connected layer which outputs 2 classes. I guessed, in our case, it is the probabilities of whether the input image pair is a positive (real-real) pair or a negative (real_forgery) pair but it does not seem so. Here are my questions:

-  Where do we apply the softmax layer at this point of the execution? (At the fully connected layer point)
-  How should we interpret these "logits" results? Are these the probabilities of classes with the first index being the probability of being a real-real or real-forgery pair? If we do so, I encountered the following situation: When I trained the model on a smaller portion of CEDAR and tried the predict method, the first element of "logits", which is the result of the fully connected layer, in [ _embedding_2logits](https://github.com/dlutkaka/DeepHSV/blob/b39f40c7545129fe5a5b6d19cc298d440d1c0bef/models.py#L56) is always zero. The second element is a floating point number between 3.0 - 4.0  , which makes logits_diff variable to be a negative number. Now, this value is returned as the distance values between pairs but by using softmax we would consider these outputs as probabilities. Can you elaborate this part please?
-  Additionally, how are we doing the classification in prediction method? It seems we are just trying to compute the minimum threshold where we get FPR and FNR values equal and predict function returns the EER value we get in that case, and the threshold value we calculate. Is the predict method not actually a prediction but calculation of this threshold value and we should afterwards do predictions by using that threshold value, stating that values smaller than that threshold values are real-real and greater ones are real-forgery pair? Then again, if that threshold value is negative, how should we interpret it?

Thank you so much in advance.

Kind Regards.

**Comment by ChuangLee** [comment]
Hi。you are right, class 0 means negative pair. class 1 means postive pair

> * Where do we apply the softmax layer at this point of the execution? (At the fully connected layer point)
```
logits = layers_lib.fully_connected(
        embeddings, 2, scope='fc3', reuse=tf.AUTO_REUSE)
```
just in [_embedding_2logits](https://github.com/dlutkaka/DeepHSV/blob/b39f40c7545129fe5a5b6d19cc298d440d1c0bef/models.py#L54)

I think the logits is not interpretable, and this is the blackbox problem of deeplearning, But sometimes in actual scenario, all we need is a threshold,  it doesn't  matter the threshold value is negative or positive, It just works
